gitpod-io/gitpod
# [UBP] Allow individual users to sign up for usage based pricing

## Description

Allow individual users to sign up for usage based billing via their settings page:

https://user-images.githubusercontent.com/8225907/190082718-027dbeeb-15bc-4836-9bf8-55c35f258fde.mov

Previously, signup for usage based billing was only available to teams.

This PR builds on several previous PRs to add the necessary endpoints to the server JSON-RPC API.
* https://github.com/gitpod-io/gitpod/pull/12753
* https://github.com/gitpod-io/gitpod/pull/12876
* https://github.com/gitpod-io/gitpod/pull/12864
* https://github.com/gitpod-io/gitpod/pull/12767
* https://github.com/gitpod-io/gitpod/pull/12887

## Related Issue(s)

Part of https://github.com/gitpod-io/gitpod/issues/12685 and https://github.com/gitpod-io/gitpod/issues/12684.

## How to test

1. Create a `"Gitpod-something"` team in the preview env.
2. Sign the team up for UBP.
3. Go to the billing tab on your individual user settings page.
4. Sign yourself up for UBP.
5. See the individual user created in the Stripe dashboard with a payment method and subscription.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
